### PR TITLE
fix: Pos return payment mode issue

### PR DESCRIPTION
**Pull request by nemesis189**
**Problem :**
On making a return POS Invoice and selecting a payment mode which is not default, eg. we choose Credit Card instead of Cash(default), on submitting the payment mode gets reset to the default one.

**Solution :**
Split the function the sets the default mop for Sales anad POS Invoices and it triggers only for Sales Invoice.

Steps:
1. Create a Return against a POS Invoice
2. change the mode of payment to a non-default one
3. check the mode of payment after submission

### CODE REVIEW

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 1]
```diff
@@ -47,7 +47,10 @@ erpnext.taxes_and_totals = class TaxesAndTotals extends erpnext.payments {
 
 		if (in_list(["Sales Invoice", "POS Invoice"], this.frm.doc.doctype) && this.frm.doc.is_pos &&
 			this.frm.doc.is_return) {
-			this.update_paid_amount_for_return();
+			if (this.frm.doc.doctype == "Sales Invoice"){
```
Comment (nextchamp-saqib) [inline erpnext/public/js/controllers/taxes_and_totals.js]:
```suggestion
			if (this.frm.doc.doctype == "Sales Invoice") {
```

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 2]
```diff
@@ -747,17 +750,15 @@ erpnext.taxes_and_totals = class TaxesAndTotals extends erpnext.payments {
 				precision("base_grand_total")
 			);
 		}
-
+
```
Comment (nextchamp-saqib) [inline erpnext/public/js/controllers/taxes_and_totals.js]:
```suggestion

```

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 3]
```diff
@@ -747,17 +750,15 @@ erpnext.taxes_and_totals = class TaxesAndTotals extends erpnext.payments {
...
 		}
-
+		
 		this.frm.doc.payments.find(pay => {
 			if (pay.default) {
 				pay.amount = total_amount_to_pay;
 			} else {
 				pay.amount = 0.0
-			}
+			}
```
Comment (nextchamp-saqib) [inline erpnext/public/js/controllers/taxes_and_totals.js]:
```suggestion
			}
```

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 4]
```diff
@@ -747,17 +750,15 @@ erpnext.taxes_and_totals = class TaxesAndTotals extends erpnext.payments {
 				precision("base_grand_total")
 			);
 		}
-
+		
 		this.frm.doc.payments.find(pay => {
 			if (pay.default) {
 				pay.amount = total_amount_to_pay;
 			} else {
 				pay.amount = 0.0
```
Comment (nextchamp-saqib) [inline erpnext/public/js/controllers/taxes_and_totals.js]:
```suggestion
				pay.amount = 0.0;
```

**Comment by AfshanKhan** [comment]
@nemesis189 also check these.
https://github.com/frappe/erpnext/pull/26853
https://github.com/frappe/erpnext/pull/26884

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 5]
```diff
@@ -734,7 +737,7 @@ erpnext.taxes_and_totals = class TaxesAndTotals extends erpnext.payments {
 		}
 	}
 
-	update_paid_amount_for_return() {
+	set_total_amount_to_default_mop(){
```
Comment (AfshanKhan) [inline erpnext/public/js/controllers/taxes_and_totals.js]: In `taxes_and_total.py` there is a function `update_paid_amount_for_return()` which does the same thing. So maybe don't rename the function for consistency.

File: erpnext/controllers/taxes_and_totals.py [hunk 6]
```diff
@@ -595,7 +595,8 @@ def calculate_outstanding_amount(self):
 				self.doc.precision("outstanding_amount"))
 
 			if self.doc.doctype == 'Sales Invoice' and self.doc.get('is_pos') and self.doc.get('is_return'):
-			 	self.update_paid_amount_for_return(total_amount_to_pay)
+			 	self.set_total_amount_to_default_mop(total_amount_to_pay)
+				self.calculate_paid_amount()
```
Comment (nextchamp-saqib) [inline erpnext/controllers/taxes_and_totals.py]:
```suggestion
				self.set_total_amount_to_default_mop(total_amount_to_pay)
				self.calculate_paid_amount()
```

**Comment by manan-ft** [comment]
Hi, is this merged in v13.11?

**Pull request merged**